tyr/servers/server.py

```diff
@@ -597,74 +617,89 @@ def resolve_iam_role(self):
 
         self.IAM_ROLE_POLICIES.extend(self.GLOBAL_IAM_ROLE_POLICIES)
         self.IAM_ROLE_POLICIES = list(set(self.IAM_ROLE_POLICIES))
-        for policy_template in self.IAM_ROLE_POLICIES:
-            policy = policy_template.format(environment=self.environment)
-
-            self.log.info('Processing policy "{policy}"'.format(policy=policy))
-
-            if policy not in existing_policies:
 
-                rolePolicy = policies[policy]
+        # If managed policies exist, then use these instead of line policies:
+        if (len(self.IAM_MANAGED_POLICIES) > 0):
+            self.log.info("Adding managed policies [" + str(self.IAM_MANAGED_POLICIES).format(environment=self.environment) + "] to role [" + self.role + "]")
 
-                if rolePolicy is None:
-                    self.log.info("No policy defined for {policy}".format(
-                                  policy=policy))
-                    continue  # Go to the next policy
+            for m_policy in self.IAM_MANAGED_POLICIES:
+                #self.log.info("Attaching {m_policy}".format(m_policy=m_policy))
+                m_policy_id = m_policy.format(environment=self.environment)
+                arn = self.iam.get_user().user.arn
+                ARN = arn[arn.find('::')+2:arn.rfind(':')]
```

Comment:
seems like this is suppose to be the account_id, not the ARN.. variable name is confusing.